### PR TITLE
#818 javadoc check

### PR DESCRIPTION
**Pull request by Isammoc**
Pull request for #818

 * Add `mvn javadoc:javadoc` during "qulice" profile execution
 * Fix several javadoc HTML errors

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#877](https://github.com/yegor256/cactoos/issues/877) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#877](https://github.com/yegor256/cactoos/issues/877) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C63314D6Z/1361812796); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/org/cactoos/scalar/AndWithIndex.java [hunk 1]
```diff
@@ -42,15 +42,17 @@
...
- * // will print "| idx #0: name: Mary | idx #1: name: John |
- * idx #2: name: William | idx #3: name: Napkin " to console
+ * {@code
+ *   new UncheckedScalar<>(
+ *       new AndWithIndex(
+ *           new IterableOf<>("Mary", "John", "William", "Napkin"),
+ *           new BiFuncOf<>((text, index) ->
+ *               System.out.printf("| idx #%d: name: %s ", index, text), true)
+ *       )
+ *   ).value();
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/scalar/AndWithIndex.java]: @Isammoc Please, correct this indentation.

File: src/main/java/org/cactoos/scalar/RetryScalar.java [hunk 2]
```diff
@@ -31,15 +31,17 @@
...
+ *   new RetryScalar<>(
+ *       () -> {
+ *           if (new SecureRandom().nextDouble() > 0.3d) {
+ *           throw new IllegalArgumentException("May happen");
+ *         }
+ *         return 0;
+ *       },
+ *       5
+ *   ).value() // will try to run 5 times before throwing an exception
+ * }
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/scalar/RetryScalar.java]: @Isammoc Please, correct the indentation of the comment above (`throw` and its outlying curly braces).

File: src/main/java/org/cactoos/scalar/Reduced.java [hunk 3]
```diff
@@ -33,32 +33,36 @@
...
+ *    final String orange = new Reduced<>(
+ *            (first, last) -> last,
+ *            new IterableOf<Scalar<String>>(
+ *                () -> "Apple",
+ *                () -> "Banana",
+ *                () -> "Orange"
+ *            )
+ *        ).value();
+ * }
  * </pre>
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/scalar/Reduced.java]: @Isammoc Please, correct the indentation of the comment above.

File: src/main/java/org/cactoos/scalar/AndWithIndex.java (hunk 1 above)
Comment (Isammoc) [reply src/main/java/org/cactoos/scalar/AndWithIndex.java]: @paulodamaso Done

File: src/main/java/org/cactoos/scalar/RetryScalar.java (hunk 2 above)
Comment (Isammoc) [reply src/main/java/org/cactoos/scalar/RetryScalar.java]: @paulodamaso Done

File: src/main/java/org/cactoos/scalar/Reduced.java (hunk 3 above)
Comment (Isammoc) [reply src/main/java/org/cactoos/scalar/Reduced.java]: @paulodamaso Done

**Comment by paulodamaso** [comment]
@Isammoc Files in c186768 still look the same, please take a look.

**Comment by Isammoc** [comment]
@paulodamaso After another review, I edit `Reduced` and `ComparableText`. Is this better?

**Comment by paulodamaso** [comment]
@rultor good to merge

File: pom.xml [hunk 4]
```diff
@@ -216,6 +216,33 @@ The MIT License (MIT)
...
           </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <version>3.0.0</version>
+            <configuration>
+              <tags>
+                <tag>
+                  <name>todo</name>
+                  <placement>a</placement>
```
Comment (llorllale) [inline pom.xml]: @Isammoc please remove `<head>` and set `<placement>` to "X" to both the `todo` config and the `checkstyle` config. I don't want them showing up in javadocs.

**Comment by llorllale** [comment]
@Isammoc I have a request; see above

File: pom.xml (hunk 4 above)
Comment (Isammoc) [reply pom.xml]: @llorllale Done. 
IMHO, that would be better to see that a method is still a work in progress (with the todo showing in javadocs) And a version should not be tagged if a puzzle exist in the codebase.

But you are the `ARC` here ^^.

**Comment by Isammoc** [comment]
@llorllale Done

File: pom.xml (hunk 4 above)
Comment (llorllale) [reply pom.xml]: @Isammoc that's ok, any issues raised would be duplicates of existing `@todo`s.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#877](https://github.com/yegor256/cactoos/issues/877) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@Isammoc According to our [Policy](http://www.zerocracy.com/policy.html#42)
> All commit messages have links to issues

Please, confirm that you'll include such links to all commits in future.

**Comment by Isammoc** [comment]
@elenavolokhova As in #876, I confirm

**Comment by elenavolokhova** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**